src/main/java/carpet/script/api/WorldAccess.java

```diff
@@ -673,7 +683,122 @@ else if (v instanceof EntityValue)
             });
             return (cc, tt) -> result[0];
         });
+        // =======================================
+        expression.addContextFunction("structure_block_save", -1, (c, t, lv) -> {
+            CarpetContext cc = (CarpetContext) c;
+            ServerWorld lv2 = cc.s.getWorld();
+
+            StructureManager lv3 = lv2.getStructureManager();
+            Structure lv6;
+            String name;
+
+            boolean returnnbt=lv.get(0).isNull();
+            Identifier struident=null;
+            if (!returnnbt){
+                name = lv.get(0).getString();
+                struident = Identifier.tryParse(name);
+                if (name.isEmpty()||name.endsWith(":")||struident == null) {
+                    return Value.NULL;
+                }
+                lv6 = lv3.getStructureOrBlank(struident);
+            }
+            else{
+                lv6= new Structure();
+            }
+            BlockArgument start = BlockArgument.findIn((CarpetContext)c, lv, 1);
+            BlockArgument ignoredblock = BlockArgument.findIn((CarpetContext)c, lv, start.offset+4,true,true,false);
+            // name,start,dimensions, ignoreEntities, ignoredBlock,~~author~~,disk
+            lv6.saveFromWorld(lv2,
+                    start.block.getPos(),
+                    new Vec3i((int) lv.get(start.offset).readInteger(), (int) lv.get(start.offset+1).readInteger(),
+                            (int) lv.get(start.offset+2).readInteger()),
+                    !lv.get(start.offset+3).getBoolean(),
+                    ignoredblock.block==null?null:ignoredblock.block.getBlockState().getBlock());
+            // lv6.setAuthor(lv.get(9).getString());//MC-140821
+            if (!returnnbt){
+                if (lv.get(ignoredblock.offset).getBoolean()) {
+                    return BooleanValue.of(lv3.saveStructure(struident));
+                }
+            }
+            else{
+                return new NBTSerializableValue(lv6.writeNbt(new NbtCompound()));
+            }
+
+            return Value.TRUE;
+        });
+
+        // =====================================
+        // =======================================
+        expression.addContextFunction("structure_block_load", -1, (c, t, lv) -> {
+            CarpetContext cc = (CarpetContext) c;
+            ServerWorld world = cc.s.getWorld();
+            boolean recievnbt=lv.get(0).getTypeString().equals("nbt");
```

Comment:
instanceof ?

Reply:
yes. i should use instanceof here.

and i should have some tests: if it crashes when received a wrong nbt file.